Scripts/Runtime/Data/VirtualData/VirtualData.cs

```diff
@@ -46,19 +46,28 @@ public class VirtualData<TKey, TInstance> : AbstractVirtualData
         /// </summary>
         public static readonly int MAX_ELEMENTS_PER_CHUNK = ChunkUtil.MaxElementsPerChunk<TInstance>();
 
-        internal static VirtualData<TKey, TInstance> Create(params AbstractVirtualData[] sources)
-        {
-            VirtualData<TKey, TInstance> virtualData = new VirtualData<TKey, TInstance>();
+        private const byte UNSET_RESULT_DESTINATION_TYPE = byte.MaxValue;
 
-            foreach (AbstractVirtualData source in sources)
-            {
-                virtualData.AddSource(source);
-                source.AddResultDestination(virtualData);
-            }
+        private VDResultsDestinationLookup m_ResultsDestinationLookup;
 
+        internal static VirtualData<TKey, TInstance> Create()
+        {
+            VirtualData<TKey, TInstance> virtualData = new VirtualData<TKey, TInstance>(UNSET_RESULT_DESTINATION_TYPE);
             return virtualData;
         }
 
+        internal static VirtualData<TKey, TInstance> CreateAsResultsDestination<TResultDestinationType>(TResultDestinationType resultDestinationType, AbstractVirtualData source)
+            where TResultDestinationType : Enum
+        {
+            byte value = (byte)(object)resultDestinationType;
```

Comment:
tin hat assert just in case it has a ton of elements.

```suggestion
        {
            Debug.Assert(resultDestinationType != UNSET_RESULT_DESTINATION_TYPE);
            byte value = (byte)(object)resultDestinationType;
```

Alternative you could validate that the enum type doesn't have a value defined at that slot
`Enum.IsDefined<TResultDestinationType>((TResultDestinationType)UNSET_RESULT_DESTINATION_TYPE))`

Reply:
Good call. Added TODO with link back to this.